addon/adapters/debug-adapter.js

```diff
@@ -0,0 +1,269 @@
+      return [];
+    } else if (attributeValue instanceof Object) {
+      return JSON.parse(JSON.stringify(attributeValue));
+    }
+    return attributeValue;
+  },
+
+  _nameToClass(modelName) {
+    return modelName;
+  },
+
+  /**
+    Fetches all Megamorphic model types
+    @private
+    @method getModelTypes
+    @return {Array} Array of model types
+  */
+  getModelTypes() {
+    let allModelNames = Object.keys(this.store._identityMap._map);
+    let modelNames = allModelNames.filter(
+      modelName => this.store._modelFactoryFor(modelName).class === MegamorphicModel
+    );
+
+    const modelTypes = modelNames.map(name => {
+      // we need to keep klass even though it is not technically needed/correct
+      // because both Ember inspector and Ember's generic debug adapter expect this data structure
+      // and use klass to generate objectIds and such
+      return { klass: name, name };
+    });
+    return modelTypes;
+  },
+
+  /**
+    Transforms a model's attribute name to be a human readable string
+    @private
+    @method columnNameToDesc
+    @param {String} name attribute name from a model type
+    @return {String} Human readable attribute name
+  */
+  columnNameToDesc(name) {
+    return capitalize(
+      underscore(name)
+        .replace(/_/g, ' ')
+        .trim()
+    );
+  },
+
+  /**
+    Wraps a given model type and observes changes to it.
+    @private
+    @method wrapModelType
+    @param {String} name Name of the model type
+    @return {Object} Contains the wrapped type
+    Format:
+      type: {Object} The wrapped type
+        The wrapped type has the following format:
+          name: {String} The name of the type
+          count: {Integer} The number of records available
+          columns: {Columns} An array of columns to describe the record
+          object: {String} The name of the model type
+  */
+  wrapModelType(name) {
+    const records = this.getRecords(name);
+    let typeToSend;
+
+    typeToSend = {
+      name,
+      count: get(records, 'length'),
+      columns: this.columnsForType(records),
+      // We pass in model name into the field object, even though it is a misnomer
+      // because Ember Inspector will use this value
+      object: name,
+    };
+
+    return typeToSend;
+  },
+
+  /**
+    Get the columns for a given model type
+    @public
+    @method columnsForType
+    @param {MegamorphicModel} record to get values from
+    @return {Array} An array of columns of the following format:
+     name: {String} The name of the column
+     desc: {String} Humanized description (what would show in a table column name)
+  */
+  columnsForType(record) {
+    let columns = [
+      {
+        name: 'id',
+        desc: 'Id',
+      },
+    ];
+    let count = 0;
+
+    if (!get(record, 'length')) {
+      return columns;
+    }
+
+    get(record, 'firstObject').eachAttribute(name => {
+      if (count++ > self.attributeLimit) {
```

Comment:
Even though we expect `attributeLimit` to be set by the inspector, i think it would make the code (& structure of the class) clearer if we set it ourselves in `init` with a comment.